core/src/Revolution/modSystemEvent.php

```diff
@@ -74,23 +65,22 @@ class modSystemEvent {
     public $params;
 
     /**
-     * Display a message to the user during the event.
-     *
-     * @todo Remove this; the centralized modRegistry will handle configurable
-     * logging of any kind of message or data to any repository or output
-     * context.  Use {@link modX::_log()} in the meantime.
-     * @param string $msg The message to display.
+     * modSystemEvent constructor.
+     * @param string $name Event name
      */
-    public function alert($msg) {}
+    public function __construct($name = null)
+    {
+        if ($name) {
+            $this->name = $name;
```

Comment:
Why not cast null to empty string like this: `$this->name = (string)$name` and avoid condition?

Reply:
For the same reason why I didn't do so `__construct(string $name = null)`. Soft check.

Reply:
The method signature is clear, it will always require a passing string, but type casting just will convert it to a string.

Reply:
I don't like this way. `(string)$array == 'Array'`.